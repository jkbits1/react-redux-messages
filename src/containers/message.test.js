import React from 'react'
import { shallow, mount } from 'enzyme'

import Message from './message'

it('renders correctly with message and default member', () => {
  const props = {
    messageDetails: {
      message:
        'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
      readableTime: '4:02:00 AM',
      readableDate: '9th February 2017'
    },
    messageMember: {
      email: 'Member info not yet loaded',
      avatar: ''
    }
  }
  const tree = shallow(<Message {...props} />)
  expect(tree).toMatchSnapshot()
})

it('renders correctly with message and member', () => {
  const props = {
    messageDetails: {
      message:
        'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
      readableTime: '4:02:00 AM',
      readableDate: '9th February 2017'
    },
    messageMember: {
      email: 'arosec@bbb.org',
      avatar: 'http://dummyimage.com/100x100.jpg/5fa2dd/ffffff'
    }
  }
  const tree = shallow(<Message {...props} />)
  expect(tree).toMatchSnapshot()
})

it('adds a new item', () => {
  const props = {
    messageDetails: {
      message:
        'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
      readableTime: '4:02:00 AM',
      readableDate: '9th February 2017'
    },
    messageMember: {
      email: 'arosec@bbb.org',
      avatar: 'http://dummyimage.com/100x100.jpg/5fa2dd/ffffff'
    }
  }

  const tree = shallow(<Message {...props} />)

  tree.find('li').simulate('mouseEnter')

  expect(tree.find('li')).toHaveLength(1)
  expect(
    tree
      .find('li')
      .find('div')
      .last()
      .text()
  ).toEqual('arosec@bbb.org')

  tree.find('li').simulate('mouseLeave')
  expect(
    tree
      .find('li')
      .find('div')
      .last()
      .text()
  ).toEqual(
    ' Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.'
  )
})
