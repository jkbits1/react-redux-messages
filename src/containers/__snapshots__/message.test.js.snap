// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with message and default member 1`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "border": "solid 1px black",
      "display": "flex",
      "justifyContent": "flex-start",
      "listStyle": "none",
      "padding": 10,
    }
  }
>
  <div
    style={
      Object {
        "width": 280,
      }
    }
  >
    4:02:00 AM
    <span
      style={
        Object {
          "paddingLeft": 15,
        }
      }
    >
      9th February 2017
    </span>
  </div>
  <div
    style={
      Object {
        "overflowWrap": "normal",
        "paddingLeft": 10,
        "width": 120,
      }
    }
  >
    No image available
  </div>
  <div
    style={
      Object {
        "width": 170,
      }
    }
  >
     
    Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.
  </div>
</li>
`;

exports[`renders correctly with message and member 1`] = `
<li
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "border": "solid 1px black",
      "display": "flex",
      "justifyContent": "flex-start",
      "listStyle": "none",
      "padding": 10,
    }
  }
>
  <div
    style={
      Object {
        "width": 280,
      }
    }
  >
    4:02:00 AM
    <span
      style={
        Object {
          "paddingLeft": 15,
        }
      }
    >
      9th February 2017
    </span>
  </div>
  <div
    style={
      Object {
        "width": 120,
      }
    }
  >
    <img
      alt="user avatar"
      src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff"
    />
  </div>
  <div
    style={
      Object {
        "width": 170,
      }
    }
  >
     
    Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.
  </div>
</li>
`;
